proto/storage/report_configuration.proto

```diff
@@ -78,6 +78,17 @@ message NotifierConfiguration {
     oneof notifier_config {
         EmailNotifierConfiguration email_config = 1;
     }
+
+    // oneof is only used to work around the postgres store limitation. The oneof wrapper can be removed in future
+    // without any impact on the wiring.
+    //
+    // @mandar: "It appears that the DB upserts are encountering failures when handling unset string foreign keys.
```

Comment:
That is correct

Reply:
Additionally, we would be enforcing the fk on `collection_id` since the beginning through Surabhi's PR. Therefore, there is no way to create a report configuration with `collection_id` once fk is added.